src/javascript/binary/websocket_pages/user/account/settings/limits.js

```diff
@@ -1,37 +1,32 @@
-const LimitsWS = require('./limits/limits.init').LimitsWS;
+const LimitsWS = require('./limits/limits.init');
 const Content  = require('../../../../common_functions/content').Content;
 const Client   = require('../../../../base/client').Client;
 
-const Limits = (function() {
-    const onLoad = function() {
+const Limits = (() => {
+    const onLoad = () => {
         Content.populate();
         if (Client.get('is_virtual')) {
             LimitsWS.limitsError();
             return;
         }
 
-        BinarySocket.init({
-            onmessage: function(msg) {
-                const response = JSON.parse(msg.data);
-                if (response) {
-                    const type = response.msg_type;
-                    const error = response.error;
+        BinarySocket.send({ get_limits: 1 }).then((response) => {
+            if (response) {
+                const type = response.msg_type;
+                const error = response.error;
 
-                    if (type === 'authorize' && Client.get('is_virtual')) {
-                        LimitsWS.limitsError(error);
-                    } else if (type === 'get_limits' && !error) {
-                        LimitsWS.limitsHandler(response);
-                    } else if (error) {
-                        LimitsWS.limitsError(error);
-                    }
+                if (type === 'authorize' && Client.get('is_virtual')) {
```

Comment:
There is only `get_limits` response here. For this purpose, all statements in onLoad can be wrapped inside:
```
BinarySocket.wait('authorize').then(() => {
    ...
});
```